src/js/media-controller.js

```diff
@@ -257,6 +257,9 @@ class MediaController extends MediaContainer {
       'play,pause': () => {
         this.propagateMediaState(MediaUIAttributes.MEDIA_PAUSED, this.paused);
       },
+      playing: () => {
+        this.propagateMediaState(MediaUIAttributes.MEDIA_HAS_PLAYED, !this.media?.paused);
```

Comment:
Any reason we couldn't just set this to `true`?

Reply:
This likely also helps it from being set for failed autoplay, so, 👍🏻

Reply:
> If you just set it to true it gets incorrectly set during initial setup when everything fires

I'm not sure the logic here is right. If there's no `media`, has-played will be true. That may not be the specific problem.

Reply:
The logic _should_ make sense in that scenario, since this should be functionally the same as `this.media && !this.media.paused`. Unless we want `has-played` to be true in that case?

The "initial setup" thing I'm talking about is that all of these get called for every event during initialization. If the thought is just doing this (adding a `console.log` for the screenshot), this is the behavior I'm seeing when I load the basic example.

```javascript
playing: () => {
  console.log('oh hai, playing event');
  this.propagateMediaState(MediaUIAttributes.MEDIA_HAS_PLAYED, true);
}, 
```

![CleanShot 2022-02-14 at 15 11 37](https://user-images.githubusercontent.com/794988/153962556-2784fc68-ca9a-4472-bd48-48c3c44567b5.png)

What am I missing?

Reply:
> The logic should make sense in that scenario, since this should be functionally the same as this.media && !this.media.paused

Console run this: `let foo = {}; !foo.media?.paused`
Equals true, not false, thanks the to the exclamation before everything compared to `this.media && !...`

> every event during initialization

Yeah, that's a good point. Since we're using the event handlers for setting init state, just setting `true` wouldn't work. But we need the right logic there to determine has-played. It should only ever be set to true when there is a media element, that media element has a working paused API, and that paused API is false. I would try using `=== false`.

Reply:
Ahh yeah you're totally right, good catch, probably clearer just to avoid optional chaining altogether there.

Reply:
Addressed in the poster image proposal, but if the element is rejected I'll submit that as a standalone PR. https://github.com/muxinc/media-chrome/pull/161